tree/dataframe/src/RDFInterfaceUtils.cxx

```diff
@@ -75,19 +81,30 @@ void ExploreBranch(TTree &t, std::set<std::string> &bNamesReg, ColumnNames_t &bN
       if (!prefix.empty())
          newPrefix = fullName + ".";
 
-      ExploreBranch(t, bNamesReg, bNames, subBranch, newPrefix, friendName);
+      ExploreBranch(t, bNamesReg, bNames, subBranch, newPrefix, friendName, foundLeaves, allowDuplicates);
 
-      if (t.GetBranch(fullName.c_str()))
-         UpdateList(bNamesReg, bNames, fullName, friendName);
-      else if (t.GetBranch(subBranchName.c_str()))
-         UpdateList(bNamesReg, bNames, subBranchName, friendName);
+      if (t.GetBranch(fullName.c_str())) {
+         TLeaf *leaf = t.GetBranch(fullName.c_str())->GetLeaf(fullName.c_str());
```

Comment:
Wouldn't 
```
auto br = t.GetBranch(fullName.c_str());
if (br) {
   TLeaf *leaf = br->GetLeaf(fullName.c_str());
}
```
be faster? (assuming the br->GetLeaf can take the fullname it would have to do less searching)

Reply:
right -- can it take the full name? :sweat_smile: